src/systems/systems/src/navigation/radio_altimeter.rs

```diff
@@ -0,0 +1,864 @@
+/// distance and attitude cutoffs. The pair refers to 1 transmitter system and 1 receiver system.
+/// Both of these systems include their radio wave generator/receiver respectively, their
+/// respective antenna installation, and the cables between the generator/receiver and the antenna.
+#[derive(Copy, Clone)]
+pub struct Ala52BTransceiverPair {
+    alt_above_ground_id: VariableIdentifier,
+    pitch_id: VariableIdentifier,
+    bank_id: VariableIdentifier,
+
+    transmitter: AntennaInstallation,
+    receiver: AntennaInstallation,
+
+    alt_above_ground: Length,
+    pitch: Angle,
+    bank: Angle,
+}
+
+impl Ala52BTransceiverPair {
+    const ALT_ABOVE_GROUND: &'static str = "PLANE ALT ABOVE GROUND";
+    const PITCH: &'static str = "PLANE PITCH DEGREES";
+    const BANK: &'static str = "PLANE BANK DEGREES";
+
+    pub fn new(
+        context: &mut InitContext,
+        transmitter: AntennaInstallation,
+        receiver: AntennaInstallation,
+    ) -> Self {
+        Self {
+            alt_above_ground_id: context.get_identifier(Self::ALT_ABOVE_GROUND.to_owned()),
+            pitch_id: context.get_identifier(Self::PITCH.to_owned()),
+            bank_id: context.get_identifier(Self::BANK.to_owned()),
+            transmitter,
+            receiver,
+            alt_above_ground: Length::new::<foot>(0.),
+            pitch: Angle::new::<degree>(0.),
+            bank: Angle::new::<degree>(0.),
+        }
+    }
+
+    /// Returns the height over ground of the physical transmitter antenna.
+    fn transmitter_height_over_ground(&self) -> Length {
+        let vertical_offset = self.pitch.sin() * self.transmitter.z;
+        self.alt_above_ground + vertical_offset - self.transmitter.y
+    }
+
+    /// Returns the height over ground of the physical receiver antenna.
+    fn receiver_height_over_ground(&self) -> Length {
+        let vertical_offset = self.pitch.sin() * self.receiver.z;
+        self.alt_above_ground + vertical_offset - self.receiver.y
+    }
+
+    /// Returns the direct distance between the antennas (usually along the aircraft's fuselage)
+    fn distance_between_antennas(&self) -> Length {
+        (self.transmitter.z - self.receiver.z).abs()
+    }
+
+    /// Returns the distance between the projected ground tracks of the transmitter and receiver.
+    fn antenna_along_ground_distance(&self) -> Length {
+        self.pitch.cos() * self.distance_between_antennas()
+    }
+}
+
+impl TransceiverPair for Ala52BTransceiverPair {
+    /// Returns the time the signal took to travel from one of the transceivers to the other,
+    /// to the receiver, or None if there is no path (or the signal is too weak).
+    /// While it is simplified compared to a proper frequency simulation, you can imagine that the
+    /// measured runtime is already based on the filtered difference between the sent and received
+    /// frequency in an FMCW-based radar.
```

Comment:
**suggestion**: Link to items by adding `[` and `]`. You can see the result of this particular change by executing: `cargo doc --open --document-private-items`.

```suggestion
    /// Returns the time the signal took to travel from one of the transceivers to the other,
    /// to the receiver, or [None] if there is no path (or the signal is too weak).
    /// While it is simplified compared to a proper frequency simulation, you can imagine that the
    /// measured runtime is already based on the filtered difference between the sent and received
    /// frequency in an FMCW-based radar.
```